### PR TITLE
Refactor hilbert transform

### PR DESCRIPTION
**Pull request by Armin12**

### CODE REVIEW

File: ecogvis/signal_processing/processing_data.py [hunk 1]
```diff
@@ -346,22 +346,23 @@ def high_gamma_estimation(block_path, bands_vals, new_file=''):
...
             Xch = Xch.reshape(1,-1)
             Xch = Xch.astype('float32')     # signal (nChannels,nSamples)
             X_fft_h = None
-            for ii, (bp0, bp1) in enumerate(zip(band_param_0, band_param_1)):
-                kernel = gaussian(Xch.shape[-1], rate, bp0, bp1)
-                X_analytic, X_fft_h = hilbert_transform(Xch, rate, kernel, phase=None, X_fft_h=X_fft_h)
+            for ii in np.arange(40):
+                X_analytic, X_fft_h = wavelet_transform(Xch, rate, X_fft_h=X_fft_h)
+                X_analytic = X_analytic[:,:,ii]
+                X_fft_h = X_fft_h[ii+1,:]
```
Comment (bendichter) [inline ecogvis/signal_processing/processing_data.py]: I don't think this is right. Why are you re-assigning `X_fft_h` in a loop?

File: ecogvis/signal_processing/processing_data.py [hunk 2]
```diff
@@ -264,22 +263,23 @@ def spectral_decomposition(block_path, bands_vals):
...
         start = time.time()
         for ch in np.arange(nChannels):
             Xch = lfp.data[:,ch]*1e6       # 1e6 scaling helps with numerical accuracy
             Xch = Xch.reshape(1,-1)
             Xch = Xch.astype('float32')     # signal (nChannels,nSamples)
             X_fft_h = None
-            for ii, (bp0, bp1) in enumerate(zip(band_param_0, band_param_1)):
-                kernel = gaussian(Xch.shape[-1], rate, bp0, bp1)
-                X_analytic, X_fft_h = hilbert_transform(Xch, rate, kernel, phase=None, X_fft_h=X_fft_h)
+            for ii in np.arange(40):
```
Comment (bendichter) [inline ecogvis/signal_processing/processing_data.py]: don't hard-code 40

File: ecogvis/signal_processing/processing_data.py [hunk 3]
```diff
@@ -264,22 +263,23 @@ def spectral_decomposition(block_path, bands_vals):
...
+                X_analytic, X_fft_h = wavelet_transform(Xch, rate, X_fft_h=X_fft_h)
+                X_analytic = X_analytic[:,:,ii]
+                X_fft_h = X_fft_h[ii+1,:]
```
Comment (bendichter) [inline ecogvis/signal_processing/processing_data.py]: I don't think this line is right. Why are you re-assigning `X_fft_h` in a loop?

File: ecogvis/signal_processing/processing_data.py [hunk 4]
```diff
@@ -346,22 +346,23 @@ def high_gamma_estimation(block_path, bands_vals, new_file=''):
         lfp = nwb.processing['ecephys'].data_interfaces['LFP'].electrical_series['preprocessed']
         rate = lfp.rate
 
-        nBands = len(band_param_0)
+        nBands = 40
```
Comment (bendichter) [inline ecogvis/signal_processing/processing_data.py]: Why are you hard-coding 40?

Comment (Armin12) [reply ecogvis/signal_processing/processing_data.py]: Line 72 in `process_nwb.wavelet_transform` is `log_spaced_cfs(4.0749286538265, 200, 40)` which always generates 40 bands. So, the shapes of `X_analytic` and `X_fft_h` are always `(...,...,40)` and `(41,...)`

**Pull request closed without merging**